src/gitpodify.ts

```diff
@@ -15,91 +16,97 @@ let observer: MutationObserver | undefined;
  * or on the active tab when the user clicked on the Gitpod icon.
  */
 const init = async (injectedByUserClick: boolean = false) => {
-    const configProvider = await ConfigProvider.create();
-    const config = configProvider.getConfig();
+  const configProvider = await ConfigProvider.create();
+  const config = configProvider.getConfig();
 
-    // Tell every page that looks like a Gitpod installation that the browser extension is installed!
-    const gitpodExtensionCheckElement = document.getElementById('ExtensionCheck_GitpodBrowserExtension');
-    if (gitpodExtensionCheckElement) {
-        gitpodExtensionCheckElement.innerHTML = "installed";
-        if (injectedByUserClick) {
-            window.open(config.gitpodURL);
-        }
-        // This page is a Gitpod page. We are done.
-        return;
+  // Tell every page that looks like a Gitpod installation that the browser extension is installed!
+  const gitpodExtensionCheckElement = document.getElementById(
+    "ExtensionCheck_GitpodBrowserExtension"
+  );
+  if (gitpodExtensionCheckElement) {
+    gitpodExtensionCheckElement.innerHTML = "installed";
+    if (injectedByUserClick) {
+      openLink(config.gitpodURL);
     }
+    // This page is a Gitpod page. We are done.
+    return;
+  }
 
-    // Get the first injector that canHandleCurrentPage()
-    const injectorProvider = new InjectorProvider(configProvider);
-    const injector = injectorProvider.findInjectorForCurrentPage();
-    if (!injector) {
-        // We do not have an injector for this page. We just open Gitpod.
-        if (injectedByUserClick) {
-            window.open(config.gitpodURL);
-        }
-        return
+  // Get the first injector that canHandleCurrentPage()
+  const injectorProvider = new InjectorProvider(configProvider);
+  const injector = injectorProvider.findInjectorForCurrentPage();
+  if (!injector) {
+    // We do not have an injector for this page. We just open Gitpod.
+    if (injectedByUserClick) {
+      openLink(config.gitpodURL);
     }
+    return;
+  }
 
-    if (injectedByUserClick) {
-        // User clicked on the Gitpod extension icon. We open the Gitpod with this page as context.
-        window.open(renderGitpodUrl(config.gitpodURL));
+  //   BREAKING, TODO: Fix this
+  if (injectedByUserClick) {
+    // User clicked on the Gitpod extension icon. We open the Gitpod with this page as context.
+    try {
+      const currentTab = await browser.tabs.getCurrent();
+      openLink(`${config.gitpodURL}/#${currentTab.url}`);
+    } catch (e) {
+      console.error(e)
+      openLink(config.gitpodURL);
     }
```

Comment:
It looks like still the browser action (clicking on the Gitpod logo from the menu bar) only opens gitpod.io, not the repo in Gitpod. I have been trying to debug it, but so far unsuccessful - really not sure what has gone wrong.

@Siddhant-K-code do we have a good way of telling if the `gitpodify.js` is running and if it is, somehow debug it? 👀

Reply:
> It looks like still the browser action (clicking on the Gitpod logo from the menu bar) only opens gitpod.io, not the repo in Gitpod. I have been trying to debug it, but so far unsuccessful - really not sure what has gone wrong.

In the same boat ⛵ 😭 

> @Siddhant-K-code do we have a good way of telling if the gitpodify.js is running and if it is, somehow debug it? 👀

I've tried so many workarounds, but I still don't know what we are missing here. Didn't found any way to debug the state of loading `gitpodify.js` 😢